### ISSUE TITLE
csharp_linux_opt_native.csharp.Grpc.IntegrationTesting.SslCredentialsTest failure

### ISSUE THREAD

**Issue by y-zeng** [grpc/grpc#10577]
https://grpc-testing.appspot.com/job/gRPC_pull_requests_linux/4446/testReport/junit/(root)/csharp_linux_opt_native/csharp_Grpc_IntegrationTesting_SslCredentialsTest/
```
NUnitLite 3.6.0 (.NET 4.5)
Copyright (C) 2017 Charlie Poole

Runtime Environment
   OS Version: Unix 4.9.20.40920
  CLR Version: 4.0.30319.42000

Test Files
    /var/local/git/grpc/src/csharp/Grpc.IntegrationTesting/bin/Release/Grpc.IntegrationTesting.exe

Test Filters
    Test: Grpc.IntegrationTesting.SslCredentialsTest

D0411 08:57:02.520589 Grpc.Core.Internal.UnmanagedLibrary Attempting to load native library "/var/local/git/grpc/src/csharp/Grpc.IntegrationTesting/bin/Release/libgrpc_csharp_ext.x64.so"
D0411 08:57:02.552650 Grpc.Core.Internal.NativeExtension gRPC native library loaded successfully.
=> Grpc.IntegrationTesting.SslCredentialsTest.AuthContextIsPopulated
=> Grpc.IntegrationTesting.SslCredentialsTest.AuthenticatedClientAndServer

Errors, Failures and Warnings

1) Error : Grpc.IntegrationTesting.SslCredentialsTest.AuthContextIsPopulated
Grpc.Core.RpcException : Status(StatusCode=Unimplemented, Detail="")
  at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () [0x0000c] in <f712f98eb8e445c8918edaf595bbe465>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (System.Threading.Tasks.Task task) [0x0004e] in <f712f98eb8e445c8918edaf595bbe465>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (System.Threading.Tasks.Task task) [0x0002e] in <f712f98eb8e445c8918edaf595bbe465>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd (System.Threading.Tasks.Task task) [0x0000b] in <f712f98eb8e445c8918edaf595bbe465>:0 
  at System.Runtime.CompilerServices.TaskAwaiter`1[TResult].GetResult () [0x00000] in <f712f98eb8e445c8918edaf595bbe465>:0 
  at Grpc.IntegrationTesting.SslCredentialsTest+<AuthContextIsPopulated>c__async0.MoveNext () [0x000e6] in <6bfe124b968a4615bf0397a3507b5224>:0 
--- End of stack trace from previous location where exception was thrown ---
  at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () [0x0000c] in <f712f98eb8e445c8918edaf595bbe465>:0 
  at NUnit.Framework.Internal.ExceptionHelper.Rethrow (System.Exception exception) [0x00006] in <0e18fcb6480844ef9101c1a098e37eef>:0 
  at NUnit.Framework.Internal.AsyncInvocationRegion+AsyncTaskInvocationRegion.WaitForPendingOperationsToComplete (System.Object invocationResult) [0x00030] in <0e18fcb6480844ef9101c1a098e37eef>:0 
  at NUnit.Framework.Internal.Commands.TestMethodCommand.RunAsyncTestMethod (NUnit.Framework.Internal.TestExecutionContext context) [0x00038] in <0e18fcb6480844ef9101c1a098e37eef>:0 

2) Error : Grpc.IntegrationTesting.SslCredentialsTest.AuthenticatedClientAndServer
Grpc.Core.RpcException : Status(StatusCode=Cancelled, Detail="Cancelled")
  at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () [0x0000c] in <f712f98eb8e445c8918edaf595bbe465>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (System.Threading.Tasks.Task task) [0x0004e] in <f712f98eb8e445c8918edaf595bbe465>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (System.Threading.Tasks.Task task) [0x0002e] in <f712f98eb8e445c8918edaf595bbe465>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd (System.Threading.Tasks.Task task) [0x0000b] in <f712f98eb8e445c8918edaf595bbe465>:0 
  at System.Runtime.CompilerServices.TaskAwaiter`1[TResult].GetResult () [0x00000] in <f712f98eb8e445c8918edaf595bbe465>:0 
  at Grpc.Core.Internal.AsyncCall`2[TRequest,TResponse].UnaryCall (TRequest msg) [0x00180] in <878367047d154a76bd9ac86d81cf14a7>:0 
  at Grpc.Core.Calls.BlockingUnaryCall[TRequest,TResponse] (Grpc.Core.CallInvocationDetails`2[TRequest,TResponse] call, TRequest req) [0x00007] in <878367047d154a76bd9ac86d81cf14a7>:0 
  at Grpc.Core.DefaultCallInvoker.BlockingUnaryCall[TRequest,TResponse] (Grpc.Core.Method`2[TRequest,TResponse] method, System.String host, Grpc.Core.CallOptions options, TRequest request) [0x0000a] in <878367047d154a76bd9ac86d81cf14a7>:0 
  at Grpc.Core.Internal.InterceptingCallInvoker.BlockingUnaryCall[TRequest,TResponse] (Grpc.Core.Method`2[TRequest,TResponse] method, System.String host, Grpc.Core.CallOptions options, TRequest request) [0x00012] in <878367047d154a76bd9ac86d81cf14a7>:0 
  at Grpc.Testing.TestService+TestServiceClient.UnaryCall (Grpc.Testing.SimpleRequest request, Grpc.Core.CallOptions options) [0x00006] in <6bfe124b968a4615bf0397a3507b5224>:0 
  at Grpc.Testing.TestService+TestServiceClient.UnaryCall (Grpc.Testing.SimpleRequest request, Grpc.Core.Metadata headers, System.Nullable`1[T] deadline, System.Threading.CancellationToken cancellationToken) [0x0000e] in <6bfe124b968a4615bf0397a3507b5224>:0 
  at Grpc.IntegrationTesting.SslCredentialsTest.AuthenticatedClientAndServer () [0x00014] in <6bfe124b968a4615bf0397a3507b5224>:0 
  at (wrapper managed-to-native) System.Reflection.MonoMethod:InternalInvoke (System.Reflection.MonoMethod,object,object[],System.Exception&)
  at System.Reflection.MonoMethod.Invoke (System.Object obj, System.Reflection.BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) [0x00038] in <f712f98eb8e445c8918edaf595bbe465>:0 

Run Settings
    Number of Test Workers: 1
    Work Directory: /var/local/git/grpc
    Internal Trace: Off

Test Run Summary
  Overall result: Failed
  Test Count: 2, Passed: 0, Failed: 2, Warnings: 0, Inconclusive: 0, Skipped: 0
    Failed Tests - Failures: 0, Errors: 2, Invalid: 0
  Start time: 2017-04-11 08:57:02Z
    End time: 2017-04-11 08:57:02Z
    Duration: 0.305 seconds
```

**Comment by jtattermusch** [comment]
Not seen in >2months.

**Comment by yashykt** [comment]
seen again https://source.cloud.google.com/results/invocations/f94b00f4-0e3b-406c-96f1-e10b3c81f562/targets/github%2Fgrpc%2Frun_tests%2Fcsharp_linux_dbg_native%2Fcsharp.Grpc.IntegrationTesting.SslCredentialsTest/tests

**Comment by jtattermusch** [comment]
Very likely,  this has been already addressed in b/198765101 (The test failure reported is from Sep 3, and the fix was merged on Sep 7)
